lxc/completion.go

```diff
@@ -612,10 +609,26 @@ func (g *cmdGlobal) cmpInstanceAllDevices(instanceName string) ([]string, cobra.
 	devices := make([]string, 0, len(metadataConfiguration.Configs))
 
 	for key := range metadataConfiguration.Configs {
-		if strings.HasPrefix(key, "device-") {
-			parts := strings.Split(key, "-")
-			deviceName := parts[1]
-			devices = append(devices, deviceName)
+		if !strings.HasPrefix(key, "device-") {
+			continue
+		}
+
+		parts := strings.Split(key, "-")
+		deviceType := parts[1]
+
+		// "unix" is not a device, get the next part.
+		if deviceType == "unix" && len(parts) > 2 {
+			// The metadata API has "unix-usb", but the device type is just "usb"
+			if parts[2] == "usb" {
+				deviceType = "usb"
```

Comment:
Also a comment from #15311 that this should be fixed in the metadata API. I'm happy to address this in a follow up as I'm not sure how it will affect the documentation.

Reply:
yes please - this feels like a kludge that im not particular happy about.